libethashseal/Ethash.cpp

```diff
@@ -158,9 +158,18 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 		else
 			_t.checkChainId(-4);
 	}
-	// Unneeded as it's checked again in Executive. Keep it here since tests assume it's checked.
-	if (_ir & ImportRequirements::TransactionBasic && _t.baseGasRequired(evmSchedule(EnvInfo(_bi))) > _t.gas())
-		BOOST_THROW_EXCEPTION(OutOfGasIntrinsic());
+	if (_ir & ImportRequirements::TransactionBasic)
```

Comment:
So currently we propagate all the transactions with incorrect chainID-in-nonce (which is fine, since they are currently valid)
We can add the check for chainID-in-nonce to TransactionQueue::import() later after the Metropolis fork, then we'll stop to propagate them.